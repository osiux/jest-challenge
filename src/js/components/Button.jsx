import React from "react";
import PropTypes from "prop-types";

const Button = ({ onClick, context, children }) => (
    <button className={`btn btn-${context}`} onClick={onClick}>
        {children}
    </button>
);

Button.defaultProps = {
    context: "primary"
};

Button.propTypes = {
    onClick: PropTypes.func,
    context: PropTypes.string,
    children: PropTypes.node
};

export default Button;
