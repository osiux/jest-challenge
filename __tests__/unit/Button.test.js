import React from 'react';
import { shallow } from 'enzyme';

import Button from '../../src/js/components/Button';

describe('Button component', () => {
    it('matches snapshot', () => {
        const component = shallow(<Button />);

        expect(component).toMatchSnapshot();
    });

    it('renders context property', () => {
        const component = shallow(<Button context="danger" />);

        expect(component).toMatchSnapshot();

        expect(component.find('.btn-primary')).not.toHaveLength(1);
        expect(component.find('.btn-danger')).toHaveLength(1);
    });

    it('calls onClick function when clicked', () => {
        const mockFn = jest.fn();

        const component = shallow(<Button onClick={mockFn} />);

        expect(mockFn).not.toBeCalled();

        component.find('button').simulate('click');

        expect(mockFn).toBeCalled();
    });
});
