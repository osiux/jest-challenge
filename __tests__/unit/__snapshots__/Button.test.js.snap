// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component matches snapshot 1`] = `
<button
  className="btn btn-primary"
/>
`;

exports[`Button component renders context property 1`] = `
<button
  className="btn btn-danger"
/>
`;
